site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,115 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TextField from "@mui/material/TextField";
+import { useState, type FC } from "react";
+
+type TimeUnit = "days" | "hours";
+
+// Value should be in milliseconds or undefined. Undefined means no value.
+type DurationValue = number | undefined;
+
+type DurationFieldProps = {
+  label: string;
+  value: DurationValue;
+  onChange: (value: DurationValue) => void;
+};
+
+export const DurationField: FC<DurationFieldProps> = (props) => {
+  const { label, value, onChange } = props;
+  const [timeUnit, setTimeUnit] = useState<TimeUnit>(() => {
+    if (!value) {
+      return "hours";
+    }
+
+    return Number.isInteger(durationToDays(value)) ? "days" : "hours";
+  });
+
+  return (
+    <div
+      css={{
+        display: "flex",
+        gap: 8,
+      }}
+    >
+      <TextField
+        css={{ maxWidth: 160 }}
+        label={label}
+        value={
+          !value
+            ? ""
+            : timeUnit === "hours"
+              ? durationToHours(value)
+              : durationToDays(value)
+        }
```

Comment:
I'm not fully clear on how the `value` prop for `TextField` works when you give the component `inputProps` with a type of `number`, so I might be missing something here

Could the empty string be replaced with `0`, and if so, do you think it makes sense to make that the default value for `DurationField's` value prop? That way, the values collapse down early, and you don't have to juggle undefined and "zero-ish" values throughout the rest of the component

Reply:
Though maybe it'd be better to do it like this:
```tsx
const { label, value, onChange } = props;
const safeValue = Math.trunc(value ?? 0);
```

Reply:
I think it makes sense but at the same time, I think using `undefined` is more flexible without too much extra work. Let's say the user is updating the schedule settings and instead of passing `undefined` or `null` to the autostop value it uses 0. The system can interpret that autostop is active and should stop the workspace 0ms after it is on (in this use case specifically the BE should handle that but it is just an example). 

I also think having 0 in an empty input is not the best but it is just a personal preference. 

Wdyt?

Reply:
Going to remove the need of handling undefined 👍